### DISCUSSION TITLE
Validierung in Editierbare Felder - Eingabe ist erforderlich, auch wenn "Erforderlich" deaktiviert ist.

### DISCUSSION THREAD

**Discussion by faulgurke** [e-netze/webgis-community#160] [Q&A]
Wenn die Validierung (z.B. über Regex) aktiviert ist, ist die Eingabe erforderlich, auch wenn der Schalter "Erforderlich (required) deaktiviert ist.
Beispiel: Der Eintragende kann, muss aber keine email-Adresse angeben. Damit die Mailadresse gültig eingegeben wird, soll ein Regex Ausdruck die Eingabe validieren. Solbald ein Regex-Ausdruck eingetragen wird (oder eine minimale Länge) gibt es eine Fehlermeldung wenn keine Werte eingetragen werden - auch bei deaktiviertem "Erforderlich".
Nur wenn alle Einträge aus "Validierung" entfernt werden, gibt es keine Fehlermeldung.

**Comment by gis-eni** [comment]
Richtig. So solle es sein. Sobald du eine Validierung einstellst, wird validiert.
Erforderlich bedeutet, dass der Wert erforderlich (``required``) ist. Das eine Art der Validierung und bedeutet dass der Anwender für dieses Feld etwas eingeben muss.

**Comment by marko-jer** [comment]
Ich sehe es etwas differenzierter, wenn etwas eingegeben werden soll, dann muss es einer bestimmten Syntax entsprechen, sonst soll es leer bleiben...
regex-Beispiel für eine Email
`^$|^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$`
leer: ^$
oder: |

**Reply by faulgurke** [reply]
Wenn das Eingabefeld nicht erforderlich und im Eingabefeld nichts vorhanden ist, sollte nicht validiert werden, weil das keinen Sinn macht und nur unnötige Fehlersituationen auslöst (Gruß Andi).

**Reply by gis-eni** [reply]
Falsch. Wenn in der Validierung etwas eingestellt ist, wird validiert. Diese Logik zu ändern wäre ein Breaking Change! Die Validierung würde plötzlich nicht mehr gleich funktionieren. Aber du kannst ja im Regex sagen: Datum oder Leer.

Oder dein Datenmodell schreib ein Datum in ein Datenfeld ;)
(Gruß zurück)

**Reply by gis-eni** [reply]
PS:
``^((\d{2}\.\d{2}\.\d{4})|)$``

**Reply by faulgurke** [reply]
Danke - Hab ich schon im Regex gefixt (Wert oder Leer) (Gruß Jürgen ;-)
